azure/services/bastionhosts/spec_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bastionhosts
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+)
+
+var (
+	hostName        = "my-bastion-host"
+	hostType        = "my-type"
+	fakeBastionHost = armnetwork.BastionHost{
+		Location: &fakeAzureBastionSpec.Location,
+		Name:     &hostName,
+		SKU:      &armnetwork.SKU{},
+		Type:     &hostType,
+	}
+)
+
+func TestAzureBastionSpec_Parameters(t *testing.T) {
+	testCases := []struct {
+		name          string
+		spec          *AzureBastionSpec
+		existing      interface{}
+		expect        func(g *WithT, result interface{})
+		expectedError string
+	}{
+		{
+			name:     "error when existing host is not of BastionHost type",
+			spec:     &AzureBastionSpec{},
+			existing: struct{}{},
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeNil())
+			},
+			expectedError: "struct {} is not an armnetwork.BastionHost",
+		},
+		{
+			name:     "get result as nil when existing BastionHost is present",
+			spec:     &AzureBastionSpec{},
+			existing: fakeBastionHost,
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeNil())
+			},
+			expectedError: "",
+		},
+		{
+			name:     "get BastionHost when all values are present",
+			spec:     &fakeAzureBastionSpec,
```

Comment:
Also, I think it will be more useful if you define this `fakeAzureBastionSpec` here in this file. I know that is defined earlier in `bastionhosts_test.go` file. 

So probably you can do something like : 

```
        fakeSubnetID-1         = "my-subnet-id"
	fakePublicIPID-1       = "my-public-ip-id"
	fakeAzureBastionSpec-1 = AzureBastionSpec{
		Name:        "my-bastion",
		Location:    "westus",
		ClusterName: "my-cluster",
		SubnetID:    fakeSubnetID,
		PublicIPID:  fakePublicIPID,
	}
```

Reply:
It makes reviewing and navigating the code much easier.

Reply:
Similarly  at other places also in this PR